tools/asset-sync/assets.ps1

```diff
@@ -0,0 +1,552 @@
+    }
+}
+
+<#
+.SYNOPSIS
+Resolve which branch on the assets repo should be checked out, given an input Configuration.
+
+.DESCRIPTION 
+Determines the presence of a branch on the git repo. If the relevant auto/<service> branch does not exist, we should use main.
+
+.PARAMETER Config
+A PSCustomObject that contains an auto-parsed assets.json object from ResolveAssetsJson.
+#>
+Function ResolveTargetBranch {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config
+    )
+    $assetRepo = ResolveAssetRepoLocation -Config $Config
+    $branch = "main"
+    try {
+        Push-Location $assetRepo
+        
+        $latestCommit = git rev-parse "origin/$($Config.AssetsRepoBranch)"
+
+        if($LASTEXITCODE -eq 0) {
+            $branch = $Config.AssetsRepoBranch
+        }
+    }
+    finally {
+        Pop-Location
+    }
+
+    return $branch
+}
+
+<#
+.SYNOPSIS
+Initializes a recordings repo based on an assets.json file. 
+
+.DESCRIPTION
+This Function will NOT re-initialize a repo if it discovers the repo already ready to go.
+
+.PARAMETER Config
+A PSCustomObject that contains an auto-parsed assets.json content from ResolveAssetsJson.
+
+.PARAMETER ForceReinitialize
+Should this assets repo be renewed regardless of current status?
+#>
+Function InitializeAssetsRepo {
+    param(
+        [Parameter(Mandatory=$true)]
+        [PSCustomObject] $Config,
+        [Parameter(Mandatory=$false)]
+        [boolean] $ForceReinitialize = $false
+    )
+    $assetRepo = ResolveAssetRepoLocation -Config $Config
+    $initialized = IsAssetsRepoInitialized -Config $Config
+    $workCompleted = $false
+
+    if ($ForceReinitialize)
+    {
+        Remove-Item -Force -R "$assetRepo/*"
+        $initialized = $false
+    }
+
+    if (!$initialized){
+        try {
+            Push-Location $assetRepo
+        
+            Write-Host "git clone --no-checkout --filter=tree:0 https://github.com/$($Config.AssetsRepo) ."
+            git clone --no-checkout --filter=tree:0 https://github.com/$($Config.AssetsRepo) .
+
+            $targetPath = ResolveCheckoutPaths -Config $Config
+            $targetBranch = ResolveTargetBranch -Config $Config
+
+            Write-Host "git sparse-checkout init"
+            git sparse-checkout init
+
+            Write-Host "git sparse-checkout set $targetPath"
+            git sparse-checkout set $targetPath
+            
+            Write-Host "git checkout $($Config.SHA)"
+            git checkout $($Config.SHA)
+            
+            if($LASTEXITCODE -gt 0){
+                throw "Unable to clone to directory $assetRepo"
+            }
+            $workCompleted = $true
+        }
+        finally {
+            Pop-Location
+        }
+    }
+
+    return $workCompleted
+}
+
+<# 
+.SYNOPSIS
+Used to interrupt script flow and get user input.
+
+.PARAMETER Config
+A PSCustomObject that contains an auto-parsed assets.json object from ResolveAssetsJson.
+#>
+Function GetUserInput {
```

Comment:
What is this function used for is it just a debugging helper?

Reply:
I am abstracting `Get User Input` as a function so that I can mock it in integration tests 👍